source/jobs.c

```diff
@@ -840,15 +840,30 @@ size_t Jobs_UpdateMsg( JobCurrentStatus_t status,
     assert( ( ( size_t ) status ) < ARRAY_LENGTH( jobStatusString ) );
 
     size_t start = 0U;
+    size_t minimumBufferSize = JOBS_API_STATUS_LENGTH + jobStatusStringLengths[ status ] + CONST_STRLEN( "\"}" );
 
-    if( ( expectedVersion != NULL ) && ( expectedVersionLength > 0U ) && ( bufferSize >=
-                                                                           ( 34U + expectedVersionLength + jobStatusStringLengths[ status ] ) ) &&
-        ( jobStatusString[ status ] != NULL ) )
+    if( ( expectedVersion != NULL ) && ( expectedVersionLength > 0U ) )
+    {
+        minimumBufferSize += JOBS_API_EXPECTED_VERSION_LENGTH + expectedVersionLength;
+    }
+
+    bool writeFailed = bufferSize < minimumBufferSize;
```

Comment:
bool is not c89

Reply:
Agreed! The README mentions this library requires C99 to build. `bool` is also used in several other places.